### DISCUSSION TITLE
Definição de Tecnologia para o Backend

### DISCUSSION THREAD

**Discussion by aleDsz** [devolunteers/idvogados-api#1] [Ideas]
Fala meus queridos, turubão?

Tô querendo continuar esse projeto, mesmo que por meio de um fork, para que seja possível torná-lo real e utilizável pelas pessoas, os quais seriam beneficiados por serviços de advocacia.

Pensando nisso, hoje o projeto utiliza [Typescript](https://github.com/devolunteers/idvogados-api/tree/dev) e gostaria de saber se podemos seguir com esta linguagem ou se, vocês, tem alguma sugestão sobre qual tecnologia deveríamos utilizar para continuar e finalizar seu desenvolvimento.

**Comment by aleDsz** [comment]
@devolunteers/devolunteers gostaria de discutir isso com vocês

**Reply by 1garo** [reply]
fala meu shark, então ts é bem generalista né, entao daria bem em qualquer projeto, as langs q eu mais conheço é go e rust, mas ambas normalmente vao resolver uns problemas mais especificos quando precisa de performance e concorrencia, bla bla bla...
Baseado nos requisitos, perfomance, pelo menos agora não seria um problema, certo? eu gosto de ts.

**Comment by allesst1rbt** [comment]
Já olhei ts mas nunca cheguei a mexer nele profundamente, mas se forem usar
consigo estudar mais profundamente pra ajudar

Em qui, 30 de set de 2021 09:47, Alexandre Vardai ***@***.***>
escreveu:

**Reply by 1garo** [reply]
Eu acho tranquilo quando é alguem q vem do js, ele só adiciona tipagem e algumas abordagens diferentes na hora de codar

**Comment by vitorsouzaalmeida** [comment]
Minha ideia é manter o TypeScript mesmo e se manter em volta de JS... Com bibliotecas etc

**Comment by aleDsz** [comment]
Gostei da resposta de todos, acho que podemos seguir.
Vou sentar para analisar o backlog que tem no repositório original e tentar organizar pontos de discussão que farei com vocês <3

**Reply by aleDsz** [reply]
Já deixei criado também um JIRA: https://devolunteers.atlassian.net/

**Reply by 1garo** [reply]
eu n consegui acessar, fala q n tenho permissao

meu email é `[censurado]`

**Reply by aleDsz** [reply]
feito

**Comment by rafaelbreno** [comment]
Dependendo, como será um projeto mais direcionado à um contribuição em grupo, poderia ser utilizados linguagens com frameworks bem mantidas e padronizadas, na minha visão poderiam ser um dos seguintes:

Laravel (PHP)
Ruby no Rails(Ruby)
Ambos tem uma arquitetura de código bem estabelecida e uma documentação bem abrangente, facilitando quem fosse entrando no projeto com o tempo.

Tipo, isso acredito que seria um diferencial, ao invés de usar um Go, Rust, Haskell, etc.

**Reply by aleDsz** [reply]
O javascript não tem uma padronização? Nem com SOLID, DTO, express, etc.? (dúvida sincera)

**Reply by rafaelbreno** [reply]
> O javascript não tem uma padronização? Nem com SOLID, DTO, express, etc.? (dúvida sincera)

No caso, foi ideia de frameworks que conheço, não excluindo o ecossistema do JS, mas tem sim. Meu único "preconceito" com o JS é tempo de validade das libs, na época que eu trabalhei um pouco vi que de tempo em tempo, meio que uma lib morria por outra ter "assumido" o Spotlight. Mas seria uma boa questão para dar uma corrida atrás.

**Reply by aleDsz** [reply]
> No caso, foi ideia de frameworks que conheço, não excluindo o ecossistema do JS, mas tem sim. Meu único "preconceito" com o JS é tempo de validade das libs, na época que eu trabalhei um pouco vi que de tempo em tempo, meio que uma lib morria por outra ter "assumido" o Spotlight. Mas seria uma boa questão para dar uma corrida atrás.

Isso é em qualquer linguagem, não tem muito como fugir. Se para de dar manutenção, vai acabar morrendo e outras aparecendo.

**Reply by rafaelbreno** [reply]
Sim sim, mas com Ruby on Rails e Laravel, tem meio que um tempo bom de estrada ja. Porem, com um TS da vida, ate onde eu sei, da para a gente selecionar certinhos quais libs vamos usar para cada parte, permitindo uma liberdade maior. Vou dar uma pesquisa depois.

**Reply by aleDsz** [reply]
eu tava pensando nisso tbm. Rails teria uma grande vantagem por sua padronização e alta produtividade

**Comment by allesst1rbt** [comment]
***@***.*** ***@***.***>* não tem
acesso a Jira em *devolunteers.atlassian.net
<http://devolunteers.atlassian.net>*., aparentemente to sem acesso

Em sex., 1 de out. de 2021 às 01:01, Alexandre de Souza <
***@***.***> escreveu:

**Reply by aleDsz** [reply]
pode mandar teu e-mail na minha DM do twitter? Só pra não ficar exposto aqui kkkk

**Comment by allesst1rbt** [comment]
Apoio o TS mas n reclamo de ruby se pa se for ate em ocaml eu n reclamo vou
tentar cooperar mais focando em aprender coisas novas tbm

Em ter., 5 de out. de 2021 às 01:30, Alexandre de Souza <
***@***.***> escreveu:
